service/internal/nodecount/node_count.go

```diff
@@ -132,14 +138,15 @@ func (nc *NodeCount) cachedNodes(ctx context.Context, cr metav1.Object) (corev1.
 	return nodes, nil
 }
 
-func (nc *NodeCount) lookupNodes(ctx context.Context) (corev1.NodeList, error) {
-	// TODO we need to get rid off the controllercontext.Context but for now it should be fine.
-	cc, err := controllercontext.FromContext(ctx)
-	if err != nil {
+func (nc *NodeCount) lookupNodes(ctx context.Context, cr metav1.Object) (corev1.NodeList, error) {
+	client, err := nc.tenantClient.K8sClient(ctx, cr)
+	if tenantclient.IsNotAvailable(err) {
+		return corev1.NodeList{}, microerror.Mask(err)
+	} else if err != nil {
 		return corev1.NodeList{}, microerror.Mask(err)
 	}
-	if cc.Client.TenantCluster.K8s != nil {
-		nodes, err := cc.Client.TenantCluster.K8s.CoreV1().Nodes().List(metav1.ListOptions{})
+	if client.K8sClient() != nil {
+		nodes, err := client.K8sClient().CoreV1().Nodes().List(metav1.ListOptions{})
 		if err != nil {
 			return corev1.NodeList{}, microerror.Mask(err)
 		}
```

Comment:
I cannot comment on the lines below because of Github but I was wondering why we want to return an error in case there are no nodes. We can create a cluster without Node Pools and that should be fine. I am not sure how the logic handles that. To me it looks like we should just return an empty list. It is not common for list responses to return "not found errors". Lists can always be empty.